build/templates/validate-commits.yaml

```diff
@@ -0,0 +1,46 @@
+# This template is used to validate that the commit messages follow the Conventional Commits specification (https://www.conventionalcommits.org/en/v1.0.0/).
+# Consider placing this at the beginning of the build pipeline to ensure that the commits are valid before proceeding with longer build steps.
+steps:
+  - task: PowerShell@2
+    condition: eq(variables['Build.Reason'], 'PullRequest')
+    inputs:
+      targetType: 'inline'
+      script: |
+        # Pre-Validation Logging
+        Write-Host "Starting PR Validation..."
+        Write-Host "Source Branch: $(System.PullRequest.SourceBranch)"
+        Write-Host "Target Branch: $(System.PullRequest.TargetBranch)"
+        Write-Host "Pull Request ID: $(System.PullRequest.PullRequestId)"
+        write-Host "Repository: $(Build.Repository.Name)"
+        Write-Host "Build.SourceBranch: $(Build.SourceBranch)"
+
+        # Fetch commit range
+        Write-Host "Retrieving commits..."
+        git fetch origin
+        Write-Host "Commit Range: origin/$(System.PullRequest.TargetBranch)..origin/$(System.PullRequest.SourceBranch)"
+        $commits = git log origin/$(System.PullRequest.TargetBranch)..origin/$(System.PullRequest.SourceBranch) --pretty=format:"%H %B"
+        $commitCount = ($commits | Measure-Object).Count
+        Write-Host "Commits found: $commitCount"
+
+        # Regex pattern for Conventional Commits
+        $pattern = '^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test){1}(\([\w\.\-]+\))?(!)?: ([\w ])+([\s\S]*)'
```

Comment:
I think we should adjust the pattern to allow merge commits from Pull requests.
In Azure DevOps, the message looks like this
`Merged PR 51094: then the pull request title`
`Merged PR 1534: another pull request title`
In GitHub, the message looks like this
`Merge pull request #52 from nventive/dev/mach/implement-commit-validation`
`Merge pull request #49 from nventive/config/bk/integrate-advanced-bugsee`

Reply:
If we don't we can have issues when merging branches on which we did pull requests (like release branches, develop into main, feature branches, etc.).

Reply:
I will adjust the expression